### PR TITLE
refactor: use more py38 features

### PR DESCRIPTION
**Pull request by jfcherng**
Use `ruff` and my brain to do some py38 refactoring.

Probably it's hard to review but in my test, LSP-copilot is still working.

---

- Use "uv" to replace "pip" for performance
- Use "ruff" to replace "autoflake8/flake8/isort/black" for performance
- Use more py38 features

### CODE REVIEW

**Comment by jfcherng** [comment]
An issue is that, it will conflict with existing working branches/PRs.

**Comment by TerminalFi** [comment]
Tested, working. 

LGTM

File: plugin/commands.py [hunk 1]
```diff
@@ -104,10 +107,9 @@ def is_enabled(self, plugin: CopilotPlugin, session: Session) -> bool:  # type:
         return self._can_meet_requirement(session)
 
 
-class CopilotWindowCommand(BaseCopilotCommand, LspWindowCommand, metaclass=ABCMeta):
+class CopilotWindowCommand(BaseCopilotCommand, LspWindowCommand, ABC):
     def is_enabled(self) -> bool:
-        session = self.session()
-        if not session:
+        if not (session := self.session()):
```
[the remark this answers is not in the document]
Comment (jfcherng) [reply plugin/commands.py:112]: That's the only usage of `:=` operator. I use this coding style in all my Python project. But yeah, it's not a must. I am open to this. @TerminalFi

[the remark this answers is not in the document]
Comment (timfjord) [reply plugin/commands.py:112]: It is not critical, I don't think we should do something about it.

**Pull request merged**